perceval/rendering/pdisplay.py

```diff
@@ -258,6 +342,22 @@ def _pdisplay(bsc, **kwargs):
     return pdisplay_state_distrib(bsc, **kwargs)
 
 
+@dispatch(float)
+def _pdisplay(f, **kwargs):
+    opts_simple = {}
+    if "precision" in kwargs:
+        opts_simple["precision"] = kwargs["precision"]
+    return simple_float(f, **opts_simple)[1]
+
+
+@dispatch(complex)
+def _pdisplay(c, **kwargs):
+    opts_simple = {}
+    if "precision" in kwargs:
+        opts_simple["precision"] = kwargs["precision"]
+    return simple_complex(c, **opts_simple)[1]
+
```

Comment:
I am sorry, I forgot why were these `@dispatch(complex)` and `@dispatch(float)` implemented. Please explain it to me.

I see it replaces the implementation inside `_pdisplay()`, but why is this better? Do we want to use `_pdisplay` for simply float and complex numbers?

Reply:
`dispatch` decorator allow to change the method executed depending of the type of the parameter (you can see that there is 10 method_pdisplay in the same file, this would not be possible without the decorator dispatch) so there is actually a _pdisplay for each supported type.
You can read more about it here:
https://www.geeksforgeeks.org/dispatch-decorator-in-python/

Reply:
Perhaps I did not ask correctly. I wanted to know the reason behind the dispatcher for `float` and `complex`. do we really need a special one just for them?